modules/ROOT/pages/batch-reference.adoc

```diff
@@ -0,0 +1,241 @@
+
+image::batch-aggregator-fields-studio.png[alt=Batch Aggregator fields in Studio,width=500px]
+
+The following table describes each field and includes the XML element for the field.
+
+[%header, cols="1a,1a,3a"]
+|===
+| Field Name | XML | Description
+
+| Display Name |  `doc:name` | Configurable name for the component. Example: `doc:name="My Aggregator"`. The default name is `Batch Aggregator`.
+
+| Aggregator Size | `size` | Size of each array of records to process. There is no default size. The processor within the array must accept an array as input. The component continues pulling records from the batch stepping queue into arrays of the configured size until none are left. If the queue contains fewer records than the configured size, the component pulls those records into a smaller array for processing. You must set the Size (`size`) or Streaming (`streaming`) property but not both.
+
+When using this configuration, the component does not wait for the Batch Step component to finish processing all records in an instance or batch. Once the set number of records is available in the stepping queue, the Batch Aggregator component pulls those records into an array and processes them.
+
+| Streaming | `streaming` | When `streaming=true`, the component acts on all records in a batch job instance and returns the results in an array of a size equal to the number of  records in the instance. The batch size setting through the Batch Job component does not limit the number of records to process or return in the array. For example, assume that the aggregator returns the numeric IDs of a small batch job instance that processes only _ten_ records of the form `{"id": 1}`, `{"id": 2}`, and so on. In this case, the output array for the instance contains ten IDs, such as `[1, 6, 2, 7, 3, 8, 4, 9, 5, 10]`, even if the `blockSize` on the Batch Job component is set to a number smaller than the number of records in the instance, such as `5`.
+
+Processors can gain sequential access to the array of streamed records. Random access to the streamed records is not allowed because the aggregator does not preserve the processed records.
+
+There is no default. You must set the Size (`size`) or Streaming (`streaming`) property but not both.
+
+| Preserve Mime Types | `preserveMimeTypes` | Defaults to `false`. To preserve the MIME type of records processed within this component, set this attribute to `true`.  Processors that cannot determine the MIME type of data return output, such as `null`, or unreadable references to the payload, such as `[[B@e596d5c, [B@1c4c3b32], ...]`. Setting `preserveMimeTypes=true` is useful when this component is set to process a fixed number (`size`) of records.
```

Comment:
if Fixed Size and selecting field in payload, then must preserve to avoid returning null

Reply:
also could be an issue when different records have different MIME type;
@IvanAndresFritzler  will check